### ISSUE TITLE
sd3 + pcm generate black image

### ISSUE THREAD

**Issue by kamiLight** [G-U-N/Phased-Consistency-Model#32]
i run the Phased-Consistency-Model/code/text_to_image_sd3/sd3_test.py and generate the black image

![Image](https://github.com/user-attachments/assets/921c699e-fd2e-4094-8974-944ca1c7994f)

when i comment the line with torch.autocast("cuda"):, also get wrong results

![Image](https://github.com/user-attachments/assets/91b11238-5efd-4507-8360-56568f325ae5)

this is my env info,
Package                   Version
------------------------- ------------
absl-py                   2.1.0
accelerate                0.27.2
aiofiles                  23.2.1
aiohappyeyeballs          2.4.8
aiohttp                   3.11.13
aiosignal                 1.3.2
aistudio-sdk              0.2.6
altair                    5.2.0
annotated-types           0.6.0
antlr4-python3-runtime    4.9.3
anyio                     4.3.0
appdirs                   1.4.4
asttokens                 3.0.0
async-timeout             5.0.1
attrs                     23.2.0
audioread                 3.0.1
av                        14.1.0
babel                     2.17.0
bce-python-sdk            0.9.29
bitsandbytes              0.43.0
blinker                   1.9.0
blobfile                  3.0.0
bokeh                     3.6.3
boltons                   25.0.0
braceexpand               0.1.7
brisque                   0.0.16
certifi                   2024.2.2
cffi                      1.17.1
charset-normalizer        3.3.2
click                     8.1.7
colorama                  0.4.6
colorlog                  6.9.0
contourpy                 1.2.0
cramjam                   2.8.2
cycler                    0.12.1
datasets                  3.3.2
decord                    0.6.0
diffusers                 0.32.2
dill                      0.3.4
docker-pycreds            0.4.0
einops                    0.8.1
exceptiongroup            1.2.0
executing                 2.2.0
fast_dataindex            0.1.2
fastapi                   0.110.0
fastparquet               2024.2.0
ffmpy                     0.3.2
filelock                  3.13.1
Flask                     3.1.0
flask-babel               4.0.0
fonttools                 4.49.0
frozenlist                1.5.0
fsspec                    2024.2.0
ftfy                      6.1.3
future                    1.0.0
gitdb                     4.0.11
GitPython                 3.1.42
gradio                    3.48.0
gradio_client             0.6.1
grpcio                    1.62.1
h11                       0.14.0
h5py                      3.13.0
hf_transfer               0.1.9
httpcore                  1.0.4
httpx                     0.27.0
huggingface-hub           0.29.2
idna                      3.6
imageio                   2.37.0
importlib_metadata        7.0.2
importlib_resources       6.1.3
intervaltree              3.1.0
ipython                   8.33.0
itsdangerous              2.2.0
jedi                      0.19.2
jieba                     0.42.1
Jinja2                    3.1.3
joblib                    1.4.2
jsonschema                4.21.1
jsonschema-specifications 2023.12.1
kiwisolver                1.4.5
lazy_loader               0.4
librosa                   0.10.2.post1
libsvm-official           3.35.0
ligo-segments             1.4.0
llvmlite                  0.44.0
lpips                     0.1.4
lxml                      5.3.1
Markdown                  3.5.2
markdown-it-py            3.0.0
MarkupSafe                2.1.5
matplotlib                3.8.3
matplotlib-inline         0.1.7
mdurl                     0.1.2
mido                      1.3.3
mpmath                    1.3.0
msgpack                   1.1.0
multidict                 6.1.0
multiprocess              0.70.12.2
networkx                  3.2.1
note-seq                  0.0.5
numba                     0.61.0
numpy                     1.26.4
nvidia-cublas-cu12        12.1.3.1
nvidia-cuda-cupti-cu12    12.1.105
nvidia-cuda-nvrtc-cu12    12.1.105
nvidia-cuda-runtime-cu12  12.1.105
nvidia-cudnn-cu12         8.9.2.26
nvidia-cufft-cu12         11.0.2.54
nvidia-curand-cu12        10.3.2.106
nvidia-cusolver-cu12      11.4.5.107
nvidia-cusparse-cu12      12.1.0.106
nvidia-nccl-cu12          2.19.3
nvidia-nvjitlink-cu12     12.4.99
nvidia-nvtx-cu12          12.1.105
omegaconf                 2.3.0
onnx                      1.17.0
opencv-python             4.11.0.86
orjson                    3.9.15
packaging                 24.0
pandas                    2.2.1
parameterized             0.9.0
parso                     0.8.4
peft                      0.9.0
pexpect                   4.9.0
pillow                    10.2.0
pip                       25.0
platformdirs              4.3.6
pooch                     1.8.2
pretty_midi               0.2.10
prettytable               3.14.0
prompt_toolkit            3.0.50
propcache                 0.3.0
protobuf                  4.25.3
psutil                    5.9.8
ptyprocess                0.7.0
pure_eval                 0.2.3
pyarrow                   15.0.1
pyav                      14.2.1
pybind11                  2.13.6
pycocoevalcap             1.2
pycocotools               2.0.8
pycparser                 2.22
pycryptodome              3.21.0
pycryptodomex             3.21.0
pydantic                  2.6.3
pydantic_core             2.16.3
pydub                     0.25.1
Pygments                  2.19.1
pyparsing                 3.1.2
python-dateutil           2.9.0.post0
python-multipart          0.0.9
pytorch-fid               0.3.0
pytz                      2024.1
PyYAML                    6.0.1
rarfile                   4.2
referencing               0.33.0
regex                     2023.12.25
requests                  2.31.0
requests-mock             1.12.1
rich                      13.9.4
rpds-py                   0.18.0
safetensors               0.4.2
scikit-image              0.25.2
scikit-learn              1.6.1
scipy                     1.12.0
semantic-version          2.10.0
sentencepiece             0.2.0
sentry-sdk                1.41.0
seqeval                   1.2.2
setproctitle              1.3.3
setuptools                75.8.0
shellingham               1.5.4
six                       1.16.0
smmap                     5.0.1
sniffio                   1.3.1
sortedcontainers          2.4.0
soundfile                 0.13.1
soxr                      0.5.0.post1
stack-data                0.6.3
starlette                 0.36.3
sympy                     1.12
tensorboard               2.16.2
tensorboard-data-server   0.7.2
tensorboardX              2.6.2.2
threadpoolctl             3.5.0
tifffile                  2025.2.18
tiktoken                  0.9.0
tokenizers                0.21.0
toolz                     0.12.1
torch                     2.2.0
torchvision               0.17.0
tornado                   6.4.2
tqdm                      4.66.2
traitlets                 5.14.3
trampoline                0.1.2
transformers              4.49.0
triton                    2.2.0
typer                     0.15.1
typing_extensions         4.10.0
tzdata                    2024.1
urllib3                   2.2.1
uvicorn                   0.28.0
visualdl                  2.5.3
wandb                     0.16.4
wcwidth                   0.2.13
webdataset                0.2.86
websockets                11.0.3
Werkzeug                  3.0.1
wheel                     0.45.1
xformers                  0.0.24
xxhash                    3.5.0
xyzservices               2025.1.0
yarl                      1.18.3
zipp                      3.17.0

Thanks!

**Comment by Gforky** [comment]
Hi, try to use convert.py inside sd3 training directory to convert you saved lora weights, that works for me.

**Comment by kamiLight** [comment]
> Hi, try to use convert.py inside sd3 training directory to convert you saved lora weights, that works for me.

amazing! it works, but why should convert the lora weights to fp16

**Comment by Gforky** [comment]
> > Hi, try to use convert.py inside sd3 training directory to convert you saved lora weights, that works for me.
> 
> amazing! it works, but why should convert the lora weights to fp16

No idea, especially why halve the weights, already create an issue about this.
